src/components/Editor.tsx

```diff
@@ -104,14 +104,14 @@ export const Editor = ({
     !(themeSaveStateFetched || !templateMetadata?.isThemeMode);
 
   const progress: number =
-    90 * // @ts-expect-error adding bools is fine
+    100 * // @ts-expect-error adding bools is fine
     ((!!puckConfig +
       !!templateMetadata +
       !!document +
-      (layoutSaveStateFetched || templateMetadata?.isThemeMode) +
+      ((layoutSaveStateFetched || templateMetadata?.isThemeMode) ?? 0) +
       visualConfigurationDataFetched +
-      (themeDataFetched || !templateMetadata?.isThemeMode) +
-      (themeSaveStateFetched || !templateMetadata?.isThemeMode)) /
+      ((themeDataFetched || !templateMetadata?.isThemeMode) ?? 0) +
+      ((themeSaveStateFetched || !templateMetadata?.isThemeMode) ?? 0)) /
       7);
```

Comment:
The 7 needs to change based on themeMode too. If it's not theme mode we don't care about waiting for the themeData and blocking the layout component.

Reply:
`(themeSaveStateFetched || !templateMetadata?.isThemeMode) == (false || !false) == true` if it's not theme mode and themeData isn't loaded, so it will continue